sdcm/utils/topology_ops.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import traceback
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import TypeVar
+
+from sdcm.wait import wait_for
+
+LOGGER = logging.getLogger(__name__)
+BaseNode = TypeVar("BaseNode")
+BaseScyllaCluster = TypeVar("BaseScyllaCluster")
+Session = TypeVar("Session")
+
+
+class OperationState(Enum):
+    decommission = "DECOMMISSIONING"
+    shutdown = "shutdown"
+    normal = "NORMAL"
+
+
+@dataclass
+class NodeStatus:
+    ip_address: str
+    host_id: str
+    status: str
+    up: bool
+
+
+class FailedDecommissionOperationMonitoring:
+    """Monitor status of decommission operation after fail
+
+    If decommission command was failed by any reason but decommission
+    process still running, wait while it be finished and check operation status
+    """
+
+    def __init__(self, target_node: "BaseNode", verification_node: "BaseNode",
+                 timeout: int | float = 7200, expected_exception: tuple[Exception | BaseException, ...] | None = None):
+        self.timeout = timeout
+        self.target_node = target_node
+        self.db_cluster: "BaseScyllaCluster" = target_node.parent_cluster
+        self.target_node_ip = target_node.ip_address
+        expected_exception = expected_exception or set()
+        self.expected_exception = set(expected_exception)
+        self.verification_node = verification_node
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_type:
+            LOGGER.warning("Decommission failed with error: %s", traceback.format_exception(exc_type, exc_val, exc_tb))
+            decommission_in_progress = self.is_node_decommissioning()
+            if not decommission_in_progress:
+                self.db_cluster.verify_decommission(self.target_node)
```

Comment:
`verify_decommission` terminates the node and does bunch of sttuff - how long it takes? Maybe we could also  skip that in case of KillNemesis exception?

Reply:
I set skip any checks if KillNemesis was raised